parser/src/ast.rs

```diff
@@ -0,0 +1,100 @@
+use crate::lexer::Tok;
+use std::fmt;
+
+/// [Identifier] is used to represent variable names.
+#[derive(Debug, Eq, PartialEq)]
+pub struct Identifier {
+    pub name: String,
+}
+
+impl fmt::Display for Identifier {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", &self.name)
+    }
+}
+
+/// [Source] is the root node of the AST representing the AIR constraints file.
+#[derive(Debug, PartialEq)]
+pub struct Source(pub Vec<SourceSection>);
+
+/// Source is divided into SourceSections.
+/// There are 4 types of Source Sections:
+/// - AirDef: Name of the air constraints module.
+/// - TraceCols: Trace Columns representing columns of the execution trace.
+/// - TransitionConstraints: Transition Constraints to be enforced on the trace columns defined
+///   in the TraceCols section.
+/// - BoundaryConstraints: Boundary Constraints to be enforced on the boundaries of columns defined
+///   in the TraceCols section. Currently there are two types of boundaries, First and Last
+///   representing the first and last rows of the column.
```

Comment:
nit: I would swap the order of Boundary and Transition constraints here and in the enum below and in the struct delarations further down